### ISSUE TITLE
Html Label examples from Microsoft learn do not render correctly on either iOS or android

### ISSUE THREAD

**Issue by roughiain** [dotnet/maui#16810]
### Description

Using the examples from the [documentation](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/label#display-html) on iOS the following is rendered

![Simulator Screenshot - iPhone 14 Pro Max - 2023-08-17 at 08 36 56](https://github.com/dotnet/maui/assets/19661735/583c3956-1dad-4320-a9c7-fea5719f321d)

If we trigger hot reload as per the video below

https://github.com/dotnet/maui/assets/19661735/280e4978-40bf-4e5f-b8c6-14665171bdb8

the label shows the styles per the documentation.

On android the following is rendered, hot reload does not affect the outcome.

![Screenshot_1692254890](https://github.com/dotnet/maui/assets/19661735/3390f664-b3ee-47ff-9ef3-0098ff112ee0)

### Steps to Reproduce

1. Create a new Maui App
2. Add the sample code from the [documentation](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/label#display-html)
3. Run

or clone https://github.com/roughiain/MauiHtmlLabeliOS and run

### Link to public reproduction project repository

https://github.com/roughiain/MauiHtmlLabeliOS

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS 16.4 Android api-34

### Did you find any workaround?

no

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 1.0. Repro on Windows 11, Android 13.0-API33 and iOS **.NET 8** with below Project:
[MauiHtmlLabelPlatformSpike.zip](https://github.com/dotnet/maui/files/12434906/MauiHtmlLabelPlatformSpike.zip)
Windows:
![image](https://github.com/dotnet/maui/assets/97647558/fc1252e4-f219-4339-856d-9b79ec13436b)
Android:
![image](https://github.com/dotnet/maui/assets/97647558/ccf77126-2529-4462-bd05-e2893dddfb4c)
iOS:
![Style](https://github.com/dotnet/maui/assets/97647558/aed7b3f1-c3e3-4035-8c19-befb2a59dd7f)

**Comment by dinisvieira** [comment]
Also having this issue. I'm wondering if fixing this other [issue ](https://github.com/dotnet/maui/issues/4230)could have caused this?

**Comment by dustin-wojciechowski** [comment]
Testing on main and it looks like behavior is as expected on ios but not Android when it comes to textcolor? Will look into it.
IOS:
![Image](https://github.com/dotnet/maui/assets/89540402/1b007d4a-1468-4d32-95f0-551a123c044b)
Android:
![Image](https://github.com/dotnet/maui/assets/89540402/4d15d177-6e9b-4f29-96e8-ba2c0a198f2e)

**Comment by dustin-wojciechowski** [comment]
So it looks like the issue here is related to the documentation itself as well as an additional understanding of how IOS and Android devices render HTML. Android is a bit more restrictive, allowing style to be used only on block level elements such as `<span>` or `<p>`. 

In the Maui documentation, it was used in `<strong>` which is not supported by Android (but is by IOS, and this was fixed for Net 8). 

I have opened a [pull request](https://github.com/dotnet/docs-maui/pull/2190) in the maui-docs repo to fix the examples as well as elaborated on the note about underlying HTML tag support.

Apologies on the confusion and hopefully this will clarify using HTML for labels.
